### PR TITLE
Updated EC2 flavors to include all instance types

### PR DESCRIPTION
**Pull request by gonoph**
Using the json data provided by the open source project
[ec2instances][1], I was able to programmatically pull in all the
current instance types with their metadata.

[1]: https://github.com/powdahound/ec2instances.info

Also included is the ruby generator script and the .erb template of the flavors.rb file.

Be kind, it's my first ruby script.

Ideally, the generator script would be called to generate the flavors.rb file prior to the packaging and testing. I'm not familiar with travis and ruby in order to make that happen.

### CODE REVIEW

File: scripts/flavors.erb [hunk 1]
```diff
@@ -0,0 +1,235 @@
+require 'fog/aws/models/compute/flavor'
+
+module Fog
```
Comment (geemus) [inline scripts/flavors.erb]: It would probably be ok to leave this file out. I don't think we really need to build objects in order to do the formatting stuff. I think just filling in the templates directly from the data would probably make it easier to maintain/debug as needed. Does that make sense?

File: scripts/make_flavors.rb [hunk 2]
```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/ruby
```
Comment (geemus) [inline scripts/make_flavors.rb]: Looks like a great start. Rather than `/scripts/make_flavors`, it would probably make sense to add this as a rake task instead. I think that is more in fitting with normal Ruby paradigms anyway. Does that make sense?

File: scripts/flavors.erb (hunk 1 above)
Comment (gonoph) [reply scripts/flavors.erb]: Let me rephrase what I think you're trying to say, to make sure I understand. You're saying to not include the .erb template, but to write out a static file directly from the sorted hashed set?

In my mind, I was trying to introduce as little change as possible, probably at some point including a build step that generates the model file before the actual tests and packaging.

File: scripts/make_flavors.rb (hunk 2 above)
Comment (gonoph) [reply scripts/make_flavors.rb]: That does make sense, but I'll need to read up on the "proper way" of doing this. First attempt at ruby.

Do you see the rake task being ran as part of the build process, or as part of the installation process on the host?

Comment (geemus) [reply scripts/make_flavors.rb]: No worries, feel free to ask questions as you have them. I appreciate you working through this with me. I would imagine it would happen on the development side and/or as part of the build process, rather than during installation. So the release would have the output bundled with it, to keep things relatively simple (though it would require a new release to add now info, this is the state we are already in).
